azurecompute-arm/src/test/java/org/jclouds/azurecompute/arm/features/PublicIPAddressApiLiveTest.java

```diff
@@ -0,0 +1,127 @@
+import org.jclouds.azurecompute.arm.internal.BaseAzureComputeApiLiveTest;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.testng.Assert.assertFalse;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertNull;
+
+
+@Test(groups = "live", singleThreaded = true)
+public class PublicIPAddressApiLiveTest extends BaseAzureComputeApiLiveTest {
+
+   private final String publicIpAddressName = "myipaddress";
+   private final String subscriptionid = getSubscriptionId();
+   private String resourcegroup;
+
+   @BeforeClass
+   @Override
+   public void setup() {
+      super.setup();
+      resourcegroup = getResourceGroupName();
+   }
+
+   @Test(groups = "live")
+   public void deletePublicIPAddressResourceDoesNotExist() {
+      final PublicIPAddressApi ipApi = api.getPublicIPAddressApi(resourcegroup);
+      boolean status = ipApi.delete(publicIpAddressName);
+      assertFalse(status);
+   }
+
+   @Test(groups = "live", dependsOnMethods = "deletePublicIPAddressResourceDoesNotExist")
+   public void createPublicIPAddress() {
+
+      final PublicIPAddressApi ipApi = api.getPublicIPAddressApi(resourcegroup);
+
+      final Map<String, String> tags = ImmutableMap.of("testkey", "testvalue");
+
+      PublicIPAddressProperties properties =
+              PublicIPAddressProperties.create(null, null, "Static", 4, null, null);
+
+      PublicIPAddress ip = ipApi.createOrUpdate(publicIpAddressName, LOCATION, tags, properties);
+
+      assertNotNull(ip);
+      assertEquals(ip.name(), publicIpAddressName);
+      assertEquals(ip.location(), LOCATION);
+      assertEquals(ip.id(), String.format("/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Network/publicIPAddresses/%s", subscriptionid, resourcegroup, publicIpAddressName));
+      assertEquals(ip.tags().get("testkey"), "testvalue");
+      assertNotNull(ip.properties());
+      assertEquals(ip.properties().provisioningState(), "Updating");
+      assertNull(ip.properties().ipAddress()); // as we don't get IP address until Succeeded state
+      assertEquals(ip.properties().publicIPAllocationMethod(), "Static");
+      assertEquals(ip.properties().idleTimeoutInMinutes().intValue(), 4);
+   }
+
+   @Test(groups = "live", dependsOnMethods = "createPublicIPAddress")
+   public void getPublicIPAddress() {
+
+      final PublicIPAddressApi ipApi = api.getPublicIPAddressApi(resourcegroup);
+
+      PublicIPAddress ip = ipApi.get(publicIpAddressName);
+
+      //Poll until resource is ready to be used
+      while (ip.properties().provisioningState().equals("Updating")) {
+         try {
+            Thread.sleep(10 * 1000);
+         } catch (InterruptedException e) {
+            e.printStackTrace();
+         }
+         ip = ipApi.get(publicIpAddressName);
+      }
```

Comment:
Use the jclouds `predicates2.retry` construct. If this is a common thing to be done, consider creating the predicate and make it injectable so other classes can benefit from it.